src/mochijson2.erl

```diff
@@ -149,6 +149,43 @@ parse_decoder_options([{format, Format} | Rest], State)
   when Format =:= struct orelse Format =:= eep18 orelse Format =:= proplist ->
     parse_decoder_options(Rest, State#decoder{object_hook=Format}).
 
+
+-ifdef(map_available).
```

Comment:
It would be ideal to put this ifdef in the existing `json_encode/2` function rather than having entirely separate versions of this function.